File: fabric/src/main/java/wile/redstonepen/gametest/BasicLeverButtonGameTests.java
package wile.redstonepen.gametest;

import net.minecraft.gametest.framework.GameTest;
import net.minecraft.gametest.framework.GameTestHelper;
import wile.redstonepen.gametestcommon.BasicLeverButtonTests;

public class BasicLeverButtonGameTests
{
  private static final String EMPTY = "redstonepen:relay_activates_from_redstone";

  public BasicLeverButtonGameTests() {}

  @GameTest(template = EMPTY, timeoutTicks = 10)
  public static void leverUseWithoutItemTogglesPoweredFalseToTrue(GameTestHelper helper)
  { BasicLeverButtonTests.leverUseWithoutItemTogglesPoweredFalseToTrue(helper); }

  @GameTest(template = EMPTY, timeoutTicks = 10)
  public static void leverUseTwiceReturnsToUnpowered(GameTestHelper helper)
  { BasicLeverButtonTests.leverUseTwiceReturnsToUnpowered(helper); }

  @GameTest(template = EMPTY, timeoutTicks = 10)
  public static void buttonUseWithoutItemPressesAndPowers(GameTestHelper helper)
  { BasicLeverButtonTests.buttonUseWithoutItemPressesAndPowers(helper); }

  @GameTest(template = EMPTY, timeoutTicks = 10)
  public static void buttonUseOnPoweredButtonReturnsConsume(GameTestHelper helper)
  { BasicLeverButtonTests.buttonUseOnPoweredButtonReturnsConsume(helper); }

  @GameTest(template = EMPTY, timeoutTicks = 30)
  public static void pulseButtonRevertsAfterShortInterval(GameTestHelper helper)
  { BasicLeverButtonTests.pulseButtonRevertsAfterShortInterval(helper); }

  @GameTest(template = EMPTY, timeoutTicks = 10)
  public static void gaugeReadsZeroWhenNoSignal(GameTestHelper helper)
  { BasicLeverButtonTests.gaugeReadsZeroWhenNoSignal(helper); }

  @GameTest(template = EMPTY, timeoutTicks = 10)
  public static void gaugeReadsSignalFromAdjacentRedstoneBlock(GameTestHelper helper)
  { BasicLeverButtonTests.gaugeReadsSignalFromAdjacentRedstoneBlock(helper); }

  @GameTest(template = EMPTY, timeoutTicks = 5)
  public static void gaugeShouldCheckWeakPowerReturnsFalse(GameTestHelper helper)
  { BasicLeverButtonTests.gaugeShouldCheckWeakPowerReturnsFalse(helper); }

  @GameTest(template = EMPTY, timeoutTicks = 5)
  public static void gaugeGetStateForPlacementReturnsNonNull(GameTestHelper helper)
  { BasicLeverButtonTests.gaugeGetStateForPlacementReturnsNonNull(helper); }
}

File: neoforge/src/main/kotlin/wile/redstonepen/gametest/BasicLeverButtonGameTests.kt
package wile.redstonepen.gametest

import net.minecraft.gametest.framework.GameTest
import net.minecraft.gametest.framework.GameTestHelper
import net.neoforged.neoforge.gametest.GameTestHolder
import net.neoforged.neoforge.gametest.PrefixGameTestTemplate
import wile.redstonepen.ModConstants
import wile.redstonepen.gametestcommon.BasicLeverButtonTests

@GameTestHolder(ModConstants.MODID)
@PrefixGameTestTemplate(false)
object BasicLeverButtonGameTests {
    private const val EMPTY = "relay_activates_from_redstone"

    @JvmStatic
    @GameTest(template = EMPTY, timeoutTicks = 10)
    fun leverUseWithoutItemTogglesPoweredFalseToTrue(helper: GameTestHelper) =
        BasicLeverButtonTests.leverUseWithoutItemTogglesPoweredFalseToTrue(helper)

    @JvmStatic
    @GameTest(template = EMPTY, timeoutTicks = 10)
    fun leverUseTwiceReturnsToUnpowered(helper: GameTestHelper) =
        BasicLeverButtonTests.leverUseTwiceReturnsToUnpowered(helper)

    @JvmStatic
    @GameTest(template = EMPTY, timeoutTicks = 10)
    fun buttonUseWithoutItemPressesAndPowers(helper: GameTestHelper) =
        BasicLeverButtonTests.buttonUseWithoutItemPressesAndPowers(helper)

    @JvmStatic
    @GameTest(template = EMPTY, timeoutTicks = 10)
    fun buttonUseOnPoweredButtonReturnsConsume(helper: GameTestHelper) =
        BasicLeverButtonTests.buttonUseOnPoweredButtonReturnsConsume(helper)

    @JvmStatic
    @GameTest(template = EMPTY, timeoutTicks = 30)
    fun pulseButtonRevertsAfterShortInterval(helper: GameTestHelper) =
        BasicLeverButtonTests.pulseButtonRevertsAfterShortInterval(helper)

    @JvmStatic
    @GameTest(template = EMPTY, timeoutTicks = 10)
    fun gaugeReadsZeroWhenNoSignal(helper: GameTestHelper) =
        BasicLeverButtonTests.gaugeReadsZeroWhenNoSignal(helper)

    @JvmStatic
    @GameTest(template = EMPTY, timeoutTicks = 10)
    fun gaugeReadsSignalFromAdjacentRedstoneBlock(helper: GameTestHelper) =
        BasicLeverButtonTests.gaugeReadsSignalFromAdjacentRedstoneBlock(helper)

    @JvmStatic
    @GameTest(template = EMPTY, timeoutTicks = 5)
    fun gaugeShouldCheckWeakPowerReturnsFalse(helper: GameTestHelper) =
        BasicLeverButtonTests.gaugeShouldCheckWeakPowerReturnsFalse(helper)

    @JvmStatic
    @GameTest(template = EMPTY, timeoutTicks = 5)
    fun gaugeGetStateForPlacementReturnsNonNull(helper: GameTestHelper) =
        BasicLeverButtonTests.gaugeGetStateForPlacementReturnsNonNull(helper)
}
